test/lit.cfg

```diff
@@ -615,7 +615,7 @@ else:
             config.swift_driver_test_options,
         )
     )
-    toolchain_lib_dir = make_path(config.swift_lib_dir, 'swift', host_os)
+    toolchain_lib_dir = make_path(config.swift_lib_dir, 'swift', 'android' if kIsAndroid else host_os)
```

Comment:
This path was [just added late last year](https://github.com/swiftlang/swift/commit/9c9010e5b750#diff-a041dc49aeef2cbc59d598b3df453d06221ec2704c2c076590d5835cbb77de44R557) and broke building several macro tests natively on Android that have started using it.

Reply:
Hmm, the `host_os` should be android if we are building for android right?

Reply:
[`host_os` is parsed from the host triple](https://github.com/swiftlang/swift/blob/b13d5607d1720c276357518b99cbce7f9ad974ca/test/lit.cfg#L418), which on Android is `linux`. That heuristic works on most platforms, but not for an Android host.

Reply:
Ah, that is annoying. Thanks for the explanation!